drivers/spi/spi_ll_stm32.c

```diff
@@ -73,28 +73,14 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 	return 0;
 }
 
-static inline u16_t spi_stm32_next_tx(struct spi_stm32_data *data)
-{
-	u16_t tx_frame = SPI_STM32_TX_NOP;
-
-	if (spi_context_tx_buf_on(&data->ctx)) {
-		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
-			tx_frame = UNALIGNED_GET((u8_t *)(data->ctx.tx_buf));
-		} else {
-			tx_frame = UNALIGNED_GET((u16_t *)(data->ctx.tx_buf));
-		}
-	}
-
-	return tx_frame;
-}
-
 /* Shift a SPI frame as master. */
 static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 {
 	u16_t tx_frame;
 	u16_t rx_frame;
 
-	tx_frame = spi_stm32_next_tx(data);
+	tx_frame = (spi_context_tx_on(&data->ctx)) ?
+		UNALIGNED_GET((u8_t *)(data->ctx.tx_buf)) : SPI_STM32_TX_NOP;
```

Comment:
Are we sure `(SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8` is true here?

Reply:
Oh, I'm sorry. I introduced that when splitting out the commits. Though not important, I'll fix it.